// penpal/es5/enums.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NativeEventType = exports.NativeErrorName = exports.ErrorCode = exports.Resolution = exports.MessageType = void 0;
let MessageType;
exports.MessageType = MessageType;

(function (MessageType) {
  MessageType["Call"] = "call";
  MessageType["Reply"] = "reply";
  MessageType["Syn"] = "syn";
  MessageType["SynAck"] = "synAck";
  MessageType["Ack"] = "ack";
})(MessageType || (exports.MessageType = MessageType = {}));

let Resolution;
exports.Resolution = Resolution;

(function (Resolution) {
  Resolution["Fulfilled"] = "fulfilled";
  Resolution["Rejected"] = "rejected";
})(Resolution || (exports.Resolution = Resolution = {}));

let ErrorCode;
exports.ErrorCode = ErrorCode;

(function (ErrorCode) {
  ErrorCode["ConnectionDestroyed"] = "ConnectionDestroyed";
  ErrorCode["ConnectionTimeout"] = "ConnectionTimeout";
  ErrorCode["NoIframeSrc"] = "NoIframeSrc";
})(ErrorCode || (exports.ErrorCode = ErrorCode = {}));

let NativeErrorName;
exports.NativeErrorName = NativeErrorName;

(function (NativeErrorName) {
  NativeErrorName["DataCloneError"] = "DataCloneError";
})(NativeErrorName || (exports.NativeErrorName = NativeErrorName = {}));

let NativeEventType;
exports.NativeEventType = NativeEventType;

(function (NativeEventType) {
  NativeEventType["Message"] = "message";
})(NativeEventType || (exports.NativeEventType = NativeEventType = {}));
