// penpal/es5/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "connectToChild", {
  enumerable: true,
  get: function () {
    return _connectToChild.default;
  }
});
Object.defineProperty(exports, "connectToParent", {
  enumerable: true,
  get: function () {
    return _connectToParent.default;
  }
});
Object.defineProperty(exports, "ErrorCode", {
  enumerable: true,
  get: function () {
    return _enums.ErrorCode;
  }
});
Object.defineProperty(exports, "Connection", {
  enumerable: true,
  get: function () {
    return _types.Connection;
  }
});
Object.defineProperty(exports, "AsyncMethodReturns", {
  enumerable: true,
  get: function () {
    return _types.AsyncMethodReturns;
  }
});
Object.defineProperty(exports, "CallSender", {
  enumerable: true,
  get: function () {
    return _types.CallSender;
  }
});
Object.defineProperty(exports, "Methods", {
  enumerable: true,
  get: function () {
    return _types.Methods;
  }
});
Object.defineProperty(exports, "PenpalError", {
  enumerable: true,
  get: function () {
    return _types.PenpalError;
  }
});

var _connectToChild = _interopRequireDefault(require("./parent/connectToChild"));

var _connectToParent = _interopRequireDefault(require("./child/connectToParent"));

var _enums = require("./enums");

var _types = require("./types");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
